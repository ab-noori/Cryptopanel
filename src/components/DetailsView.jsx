import React from 'react';
import PropTypes from 'prop-types';
import millify from 'millify';
import './DetailsView.scss';

const DetailsView = (props) => {
  const {
    marketCapRank, currentPrice,
    marketCap, marketCapChange24h, high24h, low24h,
    priceChange24h, priceChangePercentage24h,
    marketCapChangePercentage24, ath, athChangePercentage,
    atl, atlChangePercentage, fullyDilutedValuation, totalVolume,
    totalSupply, circulatingSupply, maxSupply, publicInterestScore,
  } = props;

  const data = [
    { label: 'Rank', value: marketCapRank },
    { label: 'Price', value: `$ ${millify(currentPrice)}` },
    { label: 'Market Cap', value: `$ ${millify(marketCap)}` },
    { label: 'Market Cap Change', value: `$ ${millify(marketCapChange24h)} / ${millify(marketCapChangePercentage24)} %` },
    { label: '24H High Price', value: `$ ${millify(high24h)}` },
    { label: '24H Low Price', value: `$ ${millify(low24h)}` },
    { label: '24H Price Change', value: `$ ${millify(priceChange24h)} / ${millify(priceChangePercentage24h)} %` },
    { label: 'All Time High Value', value: `$ ${millify(ath)} / ${millify(athChangePercentage)} %` },
    { label: 'All Time Low Value', value: `$ ${millify(atl)} / ${millify(atlChangePercentage)} %` },
    { label: 'Delutated Valuation', value: `$ ${millify(fullyDilutedValuation)}` },
    { label: 'Total Volume', value: `$ ${millify(totalVolume)}` },
    { label: 'Total Supply', value: `$ ${millify(totalSupply)}` },
    { label: 'Circulating Supply', value: `$ ${millify(circulatingSupply)}` },
    { label: 'Maximum Supply', value: `$ ${millify(maxSupply)}` },
    { label: 'Interest Score', value: publicInterestScore },
  ];

  return (
    <div className="details-view">
      {' '}
      {/* Add the "details-view" class */}
      <ul>
        {data.map((item) => (
          <li key={item.label}>
            <span className="label">
              {' '}
              {/* Add the "label" class */}
              {item.label}
              :
            </span>
            <span className="value">
              {' '}
              {/* Add the "value" class */}
              {item.value}
            </span>
          </li>
        ))}
      </ul>
    </div>
  );
};

DetailsView.propTypes = {
  marketCapRank: PropTypes.number.isRequired,
  currentPrice: PropTypes.number.isRequired,
  marketCap: PropTypes.number.isRequired,
  marketCapChange24h: PropTypes.number.isRequired,
  high24h: PropTypes.number.isRequired,
  low24h: PropTypes.number.isRequired,
  priceChange24h: PropTypes.number.isRequired,
  priceChangePercentage24h: PropTypes.number.isRequired,
  marketCapChangePercentage24: PropTypes.number.isRequired,
  ath: PropTypes.number.isRequired,
  athChangePercentage: PropTypes.number.isRequired,
  atl: PropTypes.number.isRequired,
  atlChangePercentage: PropTypes.number.isRequired,
  fullyDilutedValuation: PropTypes.number.isRequired,
  totalVolume: PropTypes.number.isRequired,
  totalSupply: PropTypes.number.isRequired,
  circulatingSupply: PropTypes.number.isRequired,
  maxSupply: PropTypes.number.isRequired,
  publicInterestScore: PropTypes.number.isRequired,
};

export default DetailsView;
