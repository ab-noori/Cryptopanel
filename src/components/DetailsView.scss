.details-view {
  margin: 0;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
}

.details-view ul {
  margin: 0;
  padding: 0;
  width: 90%;
  border-radius: 15px;
  background-color: rgb(212, 245, 234);
}

.details-view li {
  padding: 10px 0;
  display: flex;
  flex-wrap: nowrap;
  justify-content: space-around;
  list-style-type: none;
  transition: background-color 0.3s; /* Adding a transition for smooth effect */
}

.details-view span.label {
  text-align: left;
  width: 30%;
  font-weight: bold;
  color: #333;
}

.details-view span.value {
  text-align: left;
  width: 10%;
  color: #333;
}

.details-view li:nth-child(even) {
  background-color: #f2f2f2;
}

/* Hover effect */
.details-view li:hover {
  background-color: #e0e0e0; /* Change the background color on hover */
}

@media screen and (max-width: 768px) {
  .details-view {
    padding: 1rem 0.4rem;
    align-items: center;
  }

  .details-view ul {
    width: 95%;
    align-items: center;
  }

  .details-view li {
    align-items: center;
    justify-content: space-around;
  }

  .details-view span.label {
    width: 50%;
    white-space: nowrap;
  }

  .details-view span.value {
    width: 35%;
  }
}
