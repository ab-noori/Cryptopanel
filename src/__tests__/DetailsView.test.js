import React from 'react';
import { render } from '@testing-library/react';
import DetailsView from '../components/DetailsView';

test('renders DetailsView correctly', () => {
  const props = {
    marketCapRank: 1,
    currentPrice: 1000,
    marketCap: 1000000,
    marketCapChange24h: 5000,
    high24h: 1500,
    low24h: 800,
    priceChange24h: 200,
    priceChangePercentage24h: 0.2,
    marketCapChangePercentage24: 0.5,
    ath: 2000,
    athChangePercentage: 0.4,
    atl: 500,
    atlChangePercentage: 0.1,
    fullyDilutedValuation: 2000000,
    totalVolume: 50000,
    totalSupply: 1000000,
    circulatingSupply: 800000,
    maxSupply: 2000000,
    publicInterestScore: 0.8,
  };

  /* eslint-disable react/jsx-props-no-spreading */
  const { container } = render(<DetailsView {...props} />);

  expect(container).toMatchSnapshot();
});
