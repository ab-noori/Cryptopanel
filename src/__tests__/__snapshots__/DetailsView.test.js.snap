// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DetailsView correctly 1`] = `
<div>
  <div
    class="details-view"
  >
     
    <ul>
      <li>
        <span
          class="label"
        >
           
          Rank
          :
        </span>
        <span
          class="value"
        >
           
          1
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Price
          :
        </span>
        <span
          class="value"
        >
           
          $ 1K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Market Cap
          :
        </span>
        <span
          class="value"
        >
           
          $ 1M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Market Cap Change
          :
        </span>
        <span
          class="value"
        >
           
          $ 5K / 0.5 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          24H High Price
          :
        </span>
        <span
          class="value"
        >
           
          $ 1.5K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          24H Low Price
          :
        </span>
        <span
          class="value"
        >
           
          $ 800
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          24H Price Change
          :
        </span>
        <span
          class="value"
        >
           
          $ 200 / 0.2 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          All Time High Value
          :
        </span>
        <span
          class="value"
        >
           
          $ 2K / 0.4 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          All Time Low Value
          :
        </span>
        <span
          class="value"
        >
           
          $ 500 / 0.1 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Delutated Valuation
          :
        </span>
        <span
          class="value"
        >
           
          $ 2M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Total Volume
          :
        </span>
        <span
          class="value"
        >
           
          $ 50K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Total Supply
          :
        </span>
        <span
          class="value"
        >
           
          $ 1M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Circulating Supply
          :
        </span>
        <span
          class="value"
        >
           
          $ 800K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Maximum Supply
          :
        </span>
        <span
          class="value"
        >
           
          $ 2M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Interest Score
          :
        </span>
        <span
          class="value"
        >
           
          0.8
        </span>
      </li>
    </ul>
  </div>
</div>
`;
